tools/docker/Dockerfile

```diff
@@ -0,0 +1,231 @@
+	&& cd /home/builder/go/src/github.com/containerd/containerd \
+	&& git checkout v1.2.4 \
+	&& make \
+	&& cp -R /home/builder/go/src/github.com/containerd/containerd/bin/* /output
+
+
+
+
+# Build runc
+FROM build-base as runc-build
+RUN --mount=id=gomod,target=/home/builder/go/pkg/mod,type=cache \
+	mkdir -p /home/builder/go/src/github.com/opencontainers/runc \
+	&& git clone https://github.com/opencontainers/runc /home/builder/go/src/github.com/opencontainers/runc \
+	&& cd /home/builder/go/src/github.com/opencontainers/runc \
+	&& git checkout 6635b4f0c6af3810594d2770f662f34ddc15b40d \
+	&& make static BUILDTAGS='seccomp' \
+	&& make BINDIR='/output' install
+
+
+
+
+# Build firecracker-containerd
+FROM build-base as firecracker-containerd-build
+# Normally, it would be simplest to just "ADD --chown=builder" the firecracker-containerd source in, but that results in
+# permission denied here because "ADD --chown" does not set owner recursively (when "go build" tries to create
+# binaries, it doesn't have write permission on all directories). Instead, we bind mount the firecracker-containerd src
+# directory to a tmp location and copy to one we will actually use (giving ourselves permission to it in the process).
+RUN --mount=id=gomod,target=/home/builder/go/pkg/mod,type=cache \
+	--mount=type=bind,target=_firecracker-containerd \
+	cp -R _firecracker-containerd firecracker-containerd \
+	&& cd firecracker-containerd \
+	&& make STATIC_AGENT='true' \
+	&& cp \
+		agent/agent \
+		runtime/containerd-shim-aws-firecracker \
+		snapshotter/cmd/devmapper/devmapper_snapshotter \
+		snapshotter/cmd/naive/naive_snapshotter \
+		/output
+
+
+
+
+#########################################
+#
+# VM IMAGES
+#
+#########################################
+
+
+
+# Build a rootfs for the microVM, including runc and firecracker-containerd's agent
+FROM alpine:3.8 as firecracker-vm-root
+COPY --from=runc-build /output/runc /usr/local/bin/
+COPY --from=firecracker-containerd-build /output/agent /usr/local/bin/
+ADD --chown=builder tools/docker/fc-agent.start /etc/local.d/fc-agent.start
+RUN apk add openrc \
+	&& ln -s /etc/init.d/local /etc/runlevels/default/local \
+	&& ln -s /etc/init.d/cgroups /etc/runlevels/default/cgroups \
+	&& ln -s /etc/init.d/agetty /etc/runlevels/default/agetty.ttyS0 \
+	&& ln -s /etc/init.d/devfs /etc/runlevels/boot/devfs \
+	&& ln -s /etc/init.d/hostname /etc/runlevels/boot/hostname \
+	&& ln -s /etc/init.d/procfs /etc/runlevels/boot/procfs \
+	&& ln -s /etc/init.d/sysfs /etc/runlevels/boot/sysfs
+
+# Convert the VM rootfs into an ext4 file. This step must run as root.
+FROM debian:stretch as firecracker-vm-root-builder
+COPY --from=firecracker-vm-root / /vm
+RUN mkdir -p /output \
+	&& cd /output \
+	&& mkfs.ext4 -d /vm vm.ext4 65536
+
+
+
+
+#########################################
+#
+# TEST IMAGES
+#
+#########################################
+
+
+
+# Just enough to run basic firecracker-containerd tests
```

Comment:
How come you have this one separate from firecracker-containerd-test-e2e?  Just for the visual separation?